src/main/java/com/eerussianguy/blazemap/feature/mapping/BlockColorCollector.java

```diff
@@ -24,28 +25,37 @@ public BlockColorCollector() {
     public BlockColorMD collect(Level level, int minX, int minZ, int maxX, int maxZ) {
         final int[][] colors = new int[16][16];
         final BlockColors blockColors = Minecraft.getInstance().getBlockColors();
-
+        final MutableBlockPos colorPOS = new MutableBlockPos();
 
         for(int x = 0; x < 16; x++) {
             for(int z = 0; z < 16; z++) {
                 int y = level.getHeight(Heightmap.Types.MOTION_BLOCKING, minX + x, minZ + z);
 
                 int color = -1;
                 while(color == 0 || color == -1) {
-                    POS.set(x + minX, y, z + minZ);
-                    final BlockState state = level.getBlockState(POS);
-                    color = blockColors.getColor(state, level, POS, 0);
+                    colorPOS.set(x + minX, y, z + minZ);
+                    final BlockState state = level.getBlockState(colorPOS);
+
+                    if (state.isAir()) {
```

Comment:
Added this as a "short circuit" option, as almost all of the time in my testing the first block retrieved was air.